### ISSUE TITLE
No keyboard respose

### ISSUE THREAD

**Issue by D3kss** [Microsoft/vscode#37003]
- VSCode Version: Code 1.17.2 (b813d12980308015bcd2b3a2f6efa5c810c33ba5, 2017-10-16T13:59:46.104Z)
- OS Version: Windows_NT x64 10.0.16299
- Extensions: the listing length exceeds browsers' URL characters limit
---

Steps to Reproduce:

1.No response to Keyboard input on the main editing window
2.Ther is response to keyboard inputs on extension searches

Reproduces without extensions: Yes/No

**Comment by dbaeumer** [comment]
@D3kss can you try to reproduce this with all extensions disabled. Start from cmd or powershell with `code --disable-extensions`

If you can't you need to track down the extension that causes the problem and file an issue against that extension.

**Comment by bruno-vaz** [comment]
@dbaeumer I'm having this problem, only the editor doesn't respond to character keys, copy and paste still works, delete still works.
Usually if I wait a minute or so, it will work again, and when not, I need to reload window a few times so it works.

I started Code without extensions and couldn't reproduce, I'm currently trying to track down the extension responsible.

Those are my enabled extensions at the moment:

- Auto Close Tag
- Auto Rename Tag
- Git Lens
- highlight-matching-tag
- HTML CSS Support
- Jekyll Syntax Support
- Liquid Languages Support
- Lorem ipsum
- minify
- One Dark Pro (theme)
- Sass
- SCSS IntelliSense
- twig

P.s: is there a more efficient way of tracking extensions other than enabling one by one and reloading? I'm having another problems like extensions stopping working altogether and I didn't create an issue yet because I didn't tested if it is an extension.

**Comment by dbaeumer** [comment]
@bruno-vaz unfortunately not. This being said I don't think that this is caused by an extension. Extensions run in the own process and are therefore isolated from the editor. Unless your CPU is not very high when this happens I don't think that an extension can cause such an behavior. Especially since it seems you can copy/paste and delete but you can't type new characters.

You can do the following to help use tracking this down when you see this happening:

- Open developer tools Help > Toogle Developer Tools
- Go to the performance tab
- uncheck Screenshots at the top
- you might want to do the above steps right after you opened VS Code
- Record a trace (the round button in the upper left corner)
- type in the editor
- stop the trace
- attach it here

@alexandrudima what is very interesting is that @bruno-vaz can copy/paste and delete but not type new characters.

**Comment by alexandrudima** [comment]
@dbaeumer Some extensions might think a good idea to `registerCommand('type')`. So, extensions can effectively delay typing indefinitely. One of those extensions is stealing typing in the editor.

**Comment by D3kss** [comment]
Hi


So you told me what might be the issue.


So how to fix it ?


Thanks


________________________________

**Comment by alexandrudima** [comment]
@D3kss What extensions do you have installed? Does the issue reproduce for you when disabling all extensions? We need to narrow down and find which one extension is causing this.

**Comment by D3kss** [comment]
The issue happened after i installed a lot of extensions.
So i have to disabled all and add some extensions again?

________________________________

**Comment by alexandrudima** [comment]
I have just installed all the extensions listed by @bruno-vaz and could not find one which overwrites `type`. @bruno-vaz are all your extensions up-to-date ?

@D3kss I am sorry, but we currently don't have any way to pinpoint which extension does what. Does this issue reproduce all the time for you? In all file types (e.g. `.txt`) ?

**Comment by bruno-vaz** [comment]
@dbaeumer The profile generated is attached below.
I followed the first 3 steps, then I reloaded the window, tried to type (_asd asd asd_) and stopped the recording.

[Profile-20171027T130636.zip](https://github.com/Microsoft/vscode/files/1422483/Profile-20171027T130636.zip)

**Comment by alexandrudima** [comment]
@bruno-vaz Typing immediately after starting up is delayed until the extension host finishes initialization due to a bug introduced in 1.17 where emmet blocks typing on startup. See
 https://github.com/Microsoft/vscode/issues/36575#issuecomment-338938050

Are you experiencing this problem immediately after starting up and then, once the extension host is up and running, it goes away ?

**Comment by alexandrudima** [comment]
@bruno-vaz To validate that is the root cause, can you please:
* go to the install location of VS Code `\resources\app\extensions\emmet\package.json`

And remove line 18, `"onCommand:type",`:

![image](https://user-images.githubusercontent.com/5047891/32111590-2d5e9608-bb3b-11e7-9381-9ff43c823655.png)

Does the problem reproduce after this change ?

**Comment by bruno-vaz** [comment]
@alexandrudima after commenting this line, the problem indeed goes away.
The strange thing is that sometimes it takes more than a whole minute to work again, so maybe I have another extension that takes a lot of time to start up and because of that emmet stays blocking input?

**Comment by alexandrudima** [comment]
Yes, here is what happens in 1.17:
* our built-in emmet extension added an activation event for `onCommand:type`. That's a bug, which we have fixed on master.
* this means that the first, second, etc. typing in the editor (which dispatches the command `type`) will wait for emmet to activate. Once emmet activates, typing no longer communicates or depends on the extension host.
* if the extension host process (which is shared by built-in extensions and marketplace extension) is very busy on start-up, i.e. takes 1 min to initialize, typing will be initially delayed by 1 min.
* there are 2 problems here:
  * typing should not block on emmet getting activated, we have fixed that.
  * you have installed one or more extensions that have a poor implementation, where they execute ridiculous amounts of stuff on start-up.

You can see the activation times for extensions if you run `F1 > Start-up Performance` in the `Extension Activation Stats` section. However, those times only capture the initial activation times of an extension, it very well might be that the extensions register timeouts or async work where they still make the extension host process very busy.

**Comment by bruno-vaz** [comment]
@alexandrudima thanks, so, since it's fixed on master, in the next Code update, emmet will stop blocking input?

**Comment by D3kss** [comment]
Hi

I disabled all extensions and tat resolve the issue.


Thanks


________________________________

**Comment by SebastianCApostolescu** [comment]
I'm having a similar issue with VSCode. After a while the editor stops working correctly. Sometimes it doesn't let me type anything in the editor (I can delete code, copy/paste via shortcuts, terminal woks), emmet stops working, no snippets suggestions (hanging the cursos over a function or something that props more info just displays the loading circle), built in Git doesn't work (doesn't recognize edited files, but work in terminal).

When this happens I have to 'Reload Window', which fixes everything but the typing issue (although sometime it does fix it).

Checking with @bruno-vaz 's extension list we match on the following:

- Auto Close Tag
- Auto Rename Tag
- highlight-matching-tag

Any idea on what might be causing this?

**Comment by ramya-rao-a** [comment]
@SebastianCApostolescu Can you try the latest Insiders from https://code.visualstudio.com/insiders and tell us if you still see the issue?

**Comment by alexandrudima** [comment]
As mentioned above, there are two problems:
* in 1.17, the editor doesn't accept typing until emmet is activated
* there are extensions which have poor implementations and block the extension host process for a long period of time.

The first problem is resolved on master and has shipped in the Insiders Release Channel, the second problem is not something we can fix on our side.

**Comment by D3kss** [comment]
Hi

I disabled extensions and that resolved the issue.


Thanks


________________________________

**Comment by SebastianCApostolescu** [comment]
So as suggested by @ramya-rao-a I installed the Insiders version with all the extension I have in the stable  build. At first seems that the keyboard bug it's fixed. But the other bugs that I mentioned still appears to happend, so I started disabling extensions. After disabled a few nothing really changed until I disabled Beautify and ESLint. Those extensions seems to have a big part in generating those bugs because after I disabled them it worked better (took way longer to the bugs appear).

After that I disabled all extensions and I the bugs didn't appear again. I still have to pinpoint the exact extension that causes this. But I found another issue. After a while of coding (no extensions) my CPU went to 100%.

VSCode Open.
![vscode_open](https://user-images.githubusercontent.com/21975778/32231047-dc5582d6-be54-11e7-907f-9c29ee8eb60c.PNG)

Decided to close VSCode but the CPU (after more than 5min) remain at 100%.
![vscode_closed](https://user-images.githubusercontent.com/21975778/32231116-11414f48-be55-11e7-939d-f6fcbc0d4c35.PNG)

This kinda sucks...
